src/twisted/web/twcgi.py

```diff
@@ -94,15 +96,15 @@ def render(self, request):
             env["PATH_INFO"] = "/" + "/".join(pp)
 
         if hasattr(request, "content"):
-            # request.content is either a StringIO or a TemporaryFile, and
+            # 'request.content' is either a StringIO or a TemporaryFile, and
             # the file pointer is sitting at the beginning (seek(0,0))
             request.content.seek(0,2)
             length = request.content.tell()
             request.content.seek(0,0)
             env['CONTENT_LENGTH'] = str(length)
 
         try:
-            qindex = request.uri.index('?')
+            qindex = request.uri.index(b'?')
```

Comment:
Note for the future; `twisted.python.url.URL` is the way to do this moving forward